src/reduction.jl

```diff
@@ -0,0 +1,172 @@
+            substrates[i] in rxvars || continue
+            
+            s   = substrates[i]
+            rl *= s
+            isone(stoich) && continue
+            for i in one(stoich):(stoich-one(stoich))
+                rl *= (s - i)
+            end
+            combinatoric_ratelaw && (coef *= factorial(stoich))
+        end
+        
+        !isone(coef) && (rl /= coef)
+    end
+    rl
+end
+
+"""
+    reducereaction(rx::Reaction, subs::Dict; combinatoric_ratelaw=true)
+
+Creates a reduced version of the reaction using `subs` to eliminate the chosen species.
+"""
+function reducereaction(rx::Reaction, subs::Dict; combinatoric_ratelaw=true)
+    elided_species = keys(subs)
+    
+    rate_subs = newjumpratelaw(rx; rxvars=elided_species, 
+                               combinatoric_ratelaw=combinatoric_ratelaw)
+    rate_subs = substitute(rate_subs, subs)
+    
+    subs_red, substoich_red = redspecstoich(rx.substrates, rx.substoich, elided_species)
+    prods_red, prodstoich_red = redspecstoich(rx.products, rx.prodstoich, elided_species)
+    netst_red = filter(x -> !(first(x) in elided_species), rx.netstoich)
+    
+    connection_type = rx.connection_type    
+    
+    Reaction(rate_subs, subs_red, prods_red, substoich_red, prodstoich_red, netst_red,
+             rx.only_use_rate, connection_type)
+end
+
+"""
+    reducedreactions(rn::ReactionSystem, C::AbstractMatrix, elided_species::AbstractVector, syms_cons; 
+                     combinatoric_ratelaw=true)
+
+Creates reduced versions of the reactions in the reaction system without the elided species.
+"""
+function reducedreactions(rn::ReactionSystem, C::AbstractMatrix, elided_species::AbstractVector, syms_cons; 
+                          combinatoric_ratelaw=true)
+    subs = getsubsdict(rn, C, elided_species, syms_cons)
+    map(rx -> reducereaction(rx, subs; combinatoric_ratelaw=combinatoric_ratelaw), rn.eqs)
+end
+
+"""
+    reducereacsys(rn::ReactionSystem, elided_species::AbstractVector; combinatoric_ratelaw=true)
+
+Creates reduced reaction system, eliminating the given species simultaneously. Returns a reduced
+reaction system and the matrix of conservation laws of the original system.
+
+TODO: Add support for `rn.observed` and `rn.systems`.
+"""
+function reducereacsys(rn::ReactionSystem, elided_species::AbstractVector; combinatoric_ratelaw=true)
+    S = netstoichmat(rn)
+    C = Int.(conservationlaws(S))
+    
+    @assert isempty(rn.observed)
+    @assert isempty(rn.systems)
+    
+    canelide(C, elided_species) || error("Cannot elide selected species in reaction system")
+    
+    syms_cons = conssyms(rn, C)
+    
+    states_new_idcs = filter(x -> !(x in elided_species), 1:size(C, 2))
+    states_new = map(idx -> rn.states[idx], states_new_idcs)
+    
+    ps_new = [ rn.ps; syms_cons ]
+    name_new = Symbol(rn.name, "_R")
+    
+    subs = getsubsdict(rn, C, elided_species, syms_cons)
+    reactions_new = reducedreactions(rn, C, elided_species, syms_cons; 
```

Comment:
you're going to want to add the removed species to observables with the reconstruction relations.

Reply:
Aha, I didn't know that that field was for!

Reply:
It makes so you can do things like `sol[x]`, even if `x` doesn't exist, to give the user a way to not worry about what is reduced out.